### PR TITLE
Add ability to set letsencrypt cert for service

### PR DESCRIPTION
**Pull request by AvoidMe**

### CODE REVIEW

File: modules/dynamic-apiserver/pkg/cli/install_operation.go [hunk 1]
```diff
@@ -119,6 +120,7 @@ func makeInstallCmd(k8sclient kubernetes.Interface) *cobra.Command {
 	_ = cmd.PersistentFlags().String(installDockerRegistryURL, "", "Specify private docker registry URL")
 	_ = cmd.PersistentFlags().String(installDockerRegistryUsername, "", "Specify private docker registry username")
 	_ = cmd.PersistentFlags().String(installDockerRegistryPassword, "", "Specify private docker registry password")
+	_ = cmd.PersistentFlags().String(installAdminEmailParam, "", "Specify admin email (only for letsencrypt usage)")
```
Comment (ynnt) [inline modules/dynamic-apiserver/pkg/cli/install_operation.go:123]: We should ask if a user wants to enable Let's Encrypt integration first

File: modules/dynamic-apiserver/pkg/cli/install_operation.go [hunk 2]
```diff
@@ -369,6 +378,9 @@ func runInstall(k8sclient kubernetes.Interface) func(command *cobra.Command, arg
 		}
 
 		command.Println("Installing KuberLogic...")
+		if err = usePropertyFiles(kustomizeRootDir, adminEmail, ingressClass); err != nil {
```
Comment (ynnt) [inline modules/dynamic-apiserver/pkg/cli/install_operation.go:381]: Why not viper?

File: modules/dynamic-operator/controllers/kuberlogicservice-env/env.go [hunk 3]
```diff
@@ -104,20 +107,50 @@ func (e *EnvironmentManager) SetupEnv(ctx context.Context) error {
...
 
 	if e.cfg.SvcOpts.TLSSecretName != "" {
-		if _, err := controllerruntime.CreateOrUpdate(ctx, e.Client, tlsSecret, func() error {
-			srcSecret := &v1.Secret{
+		if e.kls.Spec.UseLetsencrypt {
+			letsencryptIssuer := &certmanagerv1.ClusterIssuer{
 				ObjectMeta: v12.ObjectMeta{
-					Name:      e.cfg.SvcOpts.TLSSecretName,
-					Namespace: e.cfg.Namespace,
+					Name: "kls-letsencrypt-issuer",
```
Comment (ynnt) [inline modules/dynamic-operator/controllers/kuberlogicservice-env/env.go:113]: set issuer name via config

File: modules/dynamic-operator/controllers/kuberlogicservice-env/env.go [hunk 4]
```diff
@@ -104,20 +107,50 @@ func (e *EnvironmentManager) SetupEnv(ctx context.Context) error {
 	}
 
 	if e.cfg.SvcOpts.TLSSecretName != "" {
-		if _, err := controllerruntime.CreateOrUpdate(ctx, e.Client, tlsSecret, func() error {
-			srcSecret := &v1.Secret{
+		if e.kls.Spec.UseLetsencrypt {
+			letsencryptIssuer := &certmanagerv1.ClusterIssuer{
```
Comment (telepenin) [inline modules/dynamic-operator/controllers/kuberlogicservice-env/env.go:111]: Could you please add tests for the logic

File: modules/dynamic-apiserver/pkg/cli/install_operation.go (hunk 1 above)
Comment (AvoidMe) [reply modules/dynamic-apiserver/pkg/cli/install_operation.go:125]: Done

File: modules/dynamic-apiserver/pkg/cli/install_operation.go (hunk 2 above)
Comment (AvoidMe) [reply modules/dynamic-apiserver/pkg/cli/install_operation.go]: Done

File: modules/dynamic-operator/controllers/kuberlogicservice-env/env.go (hunk 3 above)
Comment (AvoidMe) [reply modules/dynamic-operator/controllers/kuberlogicservice-env/env.go]: Done

File: modules/dynamic-operator/controllers/kuberlogicservice-env/env.go (hunk 4 above)
Comment (AvoidMe) [reply modules/dynamic-operator/controllers/kuberlogicservice-env/env.go:111]: Done

**Review by ynnt — APPROVED** [review]

**Review by telepenin — APPROVED** [review]

**Pull request merged**